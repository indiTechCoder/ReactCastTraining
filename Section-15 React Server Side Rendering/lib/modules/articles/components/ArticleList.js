import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article'; // eslint-disable-line import/no-named-as-default

const ArticleList = ({ articles }) => {
  return (
    <div>
      {Object.values(articles).map(article => (
        <Article key={article.id} article={article} />
      ))}
    </div>
  );
};

ArticleList.propTypes = {
  articles: PropTypes.object.isRequired
};

export default ArticleList;
