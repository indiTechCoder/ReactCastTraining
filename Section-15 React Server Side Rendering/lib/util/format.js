export default class {
  static date(dateString) {
    return new Date(dateString).toDateString();
  }

  static dateTime(dateTimeString) {
    const dt = new Date(dateTimeString);
    return `${dt.toDateString()} ${dt.toLocaleTimeString()}`;
  }
}
