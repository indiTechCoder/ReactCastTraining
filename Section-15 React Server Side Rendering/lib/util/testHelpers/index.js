export { default as renderer } from 'react-test-renderer';
export { shallow, mount } from 'enzyme';
export { default as enzymeHelper } from './enzymeHelper';
export { default as snapshotHelper } from './snapshotHelper';
