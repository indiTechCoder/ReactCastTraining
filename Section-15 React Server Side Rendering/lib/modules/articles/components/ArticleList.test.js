import React from 'react';
import { shallow, enzymeHelper } from '../../../util/testHelpers';
import ArticleList from './ArticleList';

const testProps = {
  articles: {
    article1: { id: '1' },
    article2: { id: '2' }
  }
};

describe('ArticleList', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ArticleList {...testProps} />);

    enzymeHelper.assertFindCount(2, wrapper, 'Article(WithStore)');

    expect(wrapper).toMatchSnapshot();
  });
});
