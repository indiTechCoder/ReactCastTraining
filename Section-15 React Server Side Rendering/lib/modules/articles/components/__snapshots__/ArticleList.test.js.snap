// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Article(WithStore)
        article={
            Object {
                "id": "1",
              }
        }
    />
    <Article(WithStore)
        article={
            Object {
                "id": "2",
              }
        }
    />
</div>,
  "nodes": Array [
    <div>
      <Article(WithStore)
            article={
                  Object {
                        "id": "1",
                      }
            }
      />
      <Article(WithStore)
            article={
                  Object {
                        "id": "2",
                      }
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ArticleList
        articles={
                Object {
                        "article1": Object {
                          "id": "1",
                        },
                        "article2": Object {
                          "id": "2",
                        },
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "articles": Object {
            "article1": Object {
              "id": "1",
            },
            "article2": Object {
              "id": "2",
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Article(WithStore)
                    article={
                              Object {
                                        "id": "1",
                                      }
                    }
          />
          <Article(WithStore)
                    article={
                              Object {
                                        "id": "2",
                                      }
                    }
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Article(WithStore)
                    article={
                              Object {
                                        "id": "1",
                                      }
                    }
          />
          <Article(WithStore)
                    article={
                              Object {
                                        "id": "2",
                                      }
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ArticleList
    articles={
        Object {
            "article1": Object {
              "id": "1",
            },
            "article2": Object {
              "id": "2",
            },
          }
    }
/>,
}
`;
